frontend/src/components/widgets/DataFrame/columns/ProgressColumn.ts

```diff
@@ -0,0 +1,195 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  GridCell,
+  GridCellKind,
+  LoadingCell,
+} from "@glideapps/glide-data-grid"
+import { RangeCellType } from "@glideapps/glide-data-grid-cells"
+
+import { Quiver } from "src/lib/Quiver"
+import { isNullOrUndefined, notNullOrUndefined } from "src/lib/utils"
+import { isIntegerType } from "src/components/widgets/DataFrame/arrowUtils"
+
+import {
+  BaseColumn,
+  BaseColumnProps,
+  getErrorCell,
+  getEmptyCell,
+  toSafeString,
+  mergeColumnParameters,
+  formatNumber,
+  toSafeNumber,
+  countDecimals,
+} from "./utils"
+
+export interface ProgressColumnParams {
+  // The minimum permitted value. Defaults to 0.
+  readonly min_value?: number
+  // The maximum permitted value. Defaults to 100 if the underlying data is integer,
+  // or 1 for all others types.
+  readonly max_value?: number
+  // A formatting syntax (e.g. sprintf) to format the display value.
+  // This can be used for adding prefix or suffix, or changing the number of decimals of the display value.
+  readonly format?: string
+  // The stepping interval. Defaults to 0.01.
+  // Mainly useful once we provide editing capabilities.
+  readonly step?: number
+}
+
+/**
+ * A read-only column type to support rendering values that have a defined
+ * range. This is rendered via a progress-bar-like visualization.
+ */
+function ProgressColumn(props: BaseColumnProps): BaseColumn {
+  const arrowTypeName = Quiver.getTypeName(props.arrowType)
+  const isInteger = isIntegerType(arrowTypeName)
+
+  const parameters = mergeColumnParameters(
+    // Default parameters:
+    {
+      min_value: 0,
+      max_value: isInteger ? 100 : 1,
+      step: isInteger ? 1 : 0.01,
+      format: isInteger ? "%3d%%" : "percent",
+    } as ProgressColumnParams,
+    // User parameters:
+    props.columnTypeOptions
+  ) as ProgressColumnParams
+
+  // Measure the display value of the max value, so that all progress bars are aligned correctly:
+  const measureLabel = formatNumber(
+    parameters.max_value as number,
+    parameters.format
+  )
+
+  const fixedDecimals =
+    isNullOrUndefined(parameters.step) || Number.isNaN(parameters.step)
+      ? undefined
+      : countDecimals(parameters.step)
+
+  const cellTemplate = {
+    kind: GridCellKind.Custom,
+    allowOverlay: false,
+    copyData: "",
+    contentAlign: props.contentAlignment,
+    data: {
+      kind: "range-cell",
+      min: parameters.min_value,
+      max: parameters.max_value,
+      step: parameters.step,
+      value: parameters.min_value,
+      label: String(parameters.min_value),
+      measureLabel,
+      readonly: true,
+    },
+  } as RangeCellType
+
+  return {
+    ...props,
+    kind: "progress",
+    sortMode: "smart",
```

Comment:
is there a reason why we need to use smart sorting mode? If I recall, that takes a little bit more work than default.

Reply:
Smart sorting is required to handle numbers correctly (not sort numbers like strings). We also use this sort mode with `NumberColumn`. We could probably use `raw` here as well, but I haven't really tested that.